swaybar/ipc.c

```diff
@@ -322,8 +335,11 @@ void ipc_bar_init(struct bar *bar, const char *bar_id) {
 	}
 	free(res);
 	json_object_put(outputs);
-
-	const char *subscribe_json = "[ \"workspace\", \"mode\" ]";
+	// Will need to be redone for display mode toggling compatibility
+	// Will require setting bar->config->hidden_state to "show" as default
+	const char *subscribe_json = strcmp(bar->config->display_mode, "hide") == 0 ?
+		"[ \"workspace\", \"mode\", \"modifier\" ]"
+		: "[ \"workspace\", \"mode\" ]";
```

Comment:
But the IPC protocol doesn't support UNSUBSCRIBE. So if you want to allow mode switching in the future, you'll have to deal with the situation where swaybar gets events it doesn't need (consider hide->dock). Might as well implement it to begin with. In fact this way, you *avoid* having to break compat with i3 protocol, because you won't need to extend its command set.

And because i3 allows mode switching, but no unsubscribe, we can guess that this is the way i3bar handles matters as well.